mypy/subtypes.py

```diff
@@ -269,6 +270,11 @@ def is_same_type(
     )
 
 
+class _SubtypeCheck(Protocol):
+    def __call__(self, left: Type, right: Type, *, subtype_context: SubtypeContext) -> bool:
```

Comment:
I'm not too used to mypy's codebase -- is this the convention or is the convention to use mypy's extensions (which include a kwarg type for callables)

Reply:
Let's use protocols, the extensions are soft-deprecated in my view in favor of callback protocols.

Reply:
> Let's use protocols, the extensions are soft-deprecated in my view in favor of callback protocols.

Not just in your view — mypy's documentation officially states that they're deprecated :) https://mypy.readthedocs.io/en/stable/additional_features.html#extended-callable-types